doc/monitoring.md

```diff
@@ -0,0 +1,23 @@
+
+# Monitoring
+
+nDPI usually needs only a few packets per flow to get full classification and to get all the required metadata/flow_risks. After that point, nDPI stops processing the flow.
+However, in some use cases, it might be useful to allow nDPI to process the *entire* flow (i.e. *all* its packets, without any limits). Some examples:
+* to extract all the Stun metadata from a Stun flow
+* to extract all the request/replay pairs from a DNS flow
+In essence, monitoring allows the application to get the same metadata, multiple times, throughout the entire life of the session.
+
+If monitoring is enable in a flow:
+* structures `ndpi_flow->protos`, `ndpi_flow->http`, `ndpi_flow->stun`,... are populated as usual, usually with the *first* instance of the specific metadata. Nothing changed.
+* packet by packet, the new structure `ndpi_flow->monitor` is populated with the metadata of the *current* packet. This information is lost when starting processing the next packet in the same flow; it is the responsibility of the application to get it.
+
+In other words:
+* "flow metadata" is saved in `ndpi_flow->protos`, `ndpi_flow->http`, `ndpi_flow->stun`, regardless of the monitoring feature being enabled or not
+* "(curent) packet metadata" is saved in `ndpi_flow->monitor`, only if monitor is enabled
+
+Monitoring must be explicit enabled with something like: `--cfg=stun,monitoring,1`
```

Comment:
So in essence monitoring is per protocol. Any chance to use * or NULL, for all protocols?

Reply:
Note add --cfg=any,monitoring,1 to the manual

Reply:
Done